### PR TITLE
chore(src/components): Add MarkerWithLabel Component

### PR DESCRIPTION
**Pull request by d46**
I implemented MarkerWithLabel like a new component as @tomchentw said but i get some problems.

MarkerWithLabel wraps the google's Marker object so i decided to fetch prototypes from google document like in Marker macro. In this case i had to use Marker for class name to fetch Marker api.

Naming for Marker made errors like this
```javascript
Encountered two children with the same key, `Marker`.
```

I didn't touch jscodeshit side but in this implementation maybe we need to change fetching data with class name scenario.

In the constructor i had to use ReactDOM.render. MarkerWithLabel require string for labelContent.
Also when we send null or empty string to labelContent it cause crash.

I used this in componentDidUpdate shallow compare too. What do you think for this approach?

Waiting your feedback to better implementation.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -10,7 +10,7 @@ insert_final_newline = true
 
 # Matches multiple files with brace expansion notation
 # Set default charset
-[*.{js,py}]
+[*.{js,py,jsx}]
```
Comment (tomchentw) [inline .editorconfig]: Thank for this!

File: lib/components/Marker.js [hunk 2]
```diff
@@ -76,7 +76,7 @@ var Marker = (exports.Marker = (function(_React$PureComponent) {
       )
     )
 
-    var GMKlass = _this.props.markerWithLabel || google.maps.Marker
+    var GMKlass = google.maps.Marker
```
Comment (tomchentw) [inline lib/components/Marker.js]: Don't commit `lib`……yet

File: src/components/Marker.jsx [hunk 3]
```diff
@@ -318,7 +288,7 @@ export class Marker extends React.PureComponent {
    */
   constructor(props, context) {
     super(props, context)
-    const GMKlass = this.props.markerWithLabel || google.maps.Marker
+    const GMKlass = google.maps.Marker
```
Comment (tomchentw) [inline src/components/Marker.jsx]: Don't commit `src/components` yet. Should just `src/macros`

File: src/macros/Marker.jsx [hunk 4]
```diff
@@ -77,7 +52,7 @@ export class Marker extends React.PureComponent {
    */
   constructor(props, context) {
     super(props, context)
-    const GMKlass = this.props.markerWithLabel || google.maps.Marker
+    const GMKlass = google.maps.Marker
```
Comment (tomchentw) [inline src/macros/Marker.jsx]: Just inline this

File: src/macros/addons/MarkerWithLabel.jsx [hunk 5]
```diff
@@ -0,0 +1,150 @@
+/* global google */
+import React from "react"
+import PropTypes from "prop-types"
+import _MarkerWithLabel from "markerwithlabel"
```
Comment (tomchentw) [inline src/macros/addons/MarkerWithLabel.jsx]: Call it `NativeMarkerWithLabel`

File: src/macros/addons/MarkerWithLabel.jsx [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+  }
+
+  /*
+   * @url https://developers.google.com/maps/documentation/javascript/3.exp/reference#Marker
+   */
+  constructor(props, context) {
+    super(props, context)
+    const markerWithLabel = new _MarkerWithLabel(google.maps)
+    const container = document.createElement("div")
+    ReactDOM.render(this.props.children, container)
```
Comment (tomchentw) [inline src/macros/addons/MarkerWithLabel.jsx]: Don't render in the `constructor` method. Please see  the corresponding code in `InfoBox/InfoWindow`

**Comment by tomchentw** [comment]
I'll look into `Naming for Marker` issue when you fixed the issues, thanks!

**Comment by tomchentw** [comment]
@d46 I'm so sorry. Apologies for screwing up your repository and your `master` branch. The PR has been reworked to #669.

**Comment by tomchentw** [comment]
Released [v9.0.0](https://github.com/tomchentw/react-google-maps/releases/tag/v9.0.0)

**Comment by abhayjeetgupta** [comment]
@tomchentw Thanks for the MarkerwithLabel feature But I am not able to add infowindow with the marker.

**Pull request closed without merging**